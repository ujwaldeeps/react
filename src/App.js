import React,{Component} from 'react';


class App extends Component {

    constructor(props){
     super(props);
     this.state = {firstNumber: null,secondNumber:null,result:null};
    }

    summ(){
        window.alert("hello there");
    }

  render() {
    return (
        <div>
            <input type="number" max-width="50px" onChange={this.onChange}></input><br />
            <input type="number" max-width="50px" onChange={this.onChange}></input><br />
            <button onClick={this.summ} id="check">click me</button>
        </div>);
  }
}

export default App;