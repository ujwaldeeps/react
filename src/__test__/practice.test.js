import App from '../App';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import sinon from 'sinon';

it('renders correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

 it('simulates click events', () => {
    const component = shallow(<App />);
    const instance = component.instance();
    const summ = sinon.spy(instance, 'summ');
    component.find('button').simulate('click');
    expect(summ.called).toHaveBeenCalled;
  });


 it('should be selectable by id "check"', function() {
    const component = shallow(<App />);
    const button=component.find('button');
    expect((button).is('#check')).toBe(true);
  });