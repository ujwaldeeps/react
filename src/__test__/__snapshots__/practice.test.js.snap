// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    id="FirstNumber"
    max-width="50px"
    type="number"
  />
  <br />
  <input
    id="SecondNumber"
    max-width="50px"
    type="number"
  />
  <br />
  <button
    id="check"
    onClick={[Function]}
  >
    click me
  </button>
</div>
`;
